api/src/main/java/org/apache/iceberg/PartitionKey.java

```diff
@@ -36,7 +36,7 @@ public class PartitionKey implements StructLike, Serializable {
   private final PartitionSpec spec;
   private final int size;
   private final Object[] partitionTuple;
-  private final Transform[] transforms;
+  private final Function[] transforms;
```

Comment:
Should this be `Function<?, ?>` to avoid warnings about raw usage of parameterized types?

Reply:
Well, it is probably not possible because of array covariance in Java. Forget about it.